qiskit/extensions/standard/u1.py

```diff
@@ -191,3 +224,31 @@ def cu1(self, theta, control_qubit, target_qubit,
 
 
 QuantumCircuit.cu1 = cu1
+
+
+def mcu1(self, lam, control_qubits, target_qubit):
```

Comment:
With your changes to CU1Gate could we merge this function with `cu1`? If so we could put a deprecation warning on this.

Reply:
Yes, I'd like that, I'll change it 👍

Reply:
Not possible, since `cu1(theta, [0, 1], 2)` expects to create two cu1 gates, one controlled on 0 acting on 2 and one controlled on 1 acting on 2